test/test_multidevice_collectives.cpp

```diff
@@ -0,0 +1,242 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#ifdef USE_DISTRIBUTED
+#include <gtest/gtest.h>
+
+#include <multidevice/collective.h>
+#include <multidevice/communicator.h>
+#include <test/utils.h>
+
+#include <iostream>
+
+namespace nvfuser {
+
+static constexpr DeviceIdxType root = 0;
+static constexpr int tensor_size = 1024;
+static constexpr int number_of_repetitions = 8;
+
+TEST_F(NVFuserTest, FusionMultiGPU_Collective_Gather_CUDA) {
+  if (!comm.is_available() || comm.size() < 2) {
+    GTEST_SKIP() << "This test needs at least 2 ranks";
+  }
+  c10::TensorOptions options =
+      at::TensorOptions().dtype(at::kFloat).device(comm.device());
+
+  CommParams params;
+  params.root = root;
+  params.team = std::vector<DeviceIdxType>(comm.size());
+  std::iota(params.team.begin(), params.team.end(), 0);
+  params.src_bufs = {at::ones(tensor_size, options) * comm.deviceId()};
```

Comment:
Having uniform initial values could mask some errors. For example, if a recv buffer were reordered for whatever reasons, that wouldn't be detected as an error. Let's at least use something like `iota`.

Reply:
right. I did as you suggests. Now source buffers are not uniform and also changes across devices and iterations